### DISCUSSION TITLE
Task exception wasn't retrieved

### DISCUSSION THREAD

**Discussion by JohannesMaierhofer** [miguelgrinberg/microdot#204] [Q&A] [answered]
Hello, I've started working with microdot v2 and things worked out very well. In the REPL, I see the following "warning" when I send a 'GET' request. 
```
Task exception wasn't retrieved
future: <Task> coro= <generator object 'serve' at 3c188b10>
Traceback (most recent call last):
  File "asyncio/core.py", line 1, in run_until_complete
  File "microdot/microdot.py", line 1183, in serve
  File "microdot/microdot.py", line 1295, in handle_request
  File "microdot/microdot.py", line 659, in write
  File "asyncio/stream.py", line 1, in stream_awrite
  File "asyncio/stream.py", line 1, in write
TypeError: object with buffer protocol required
```
Things are working fine but I wonder what I did wrong or what I could improve?

**Comment by miguelgrinberg** [comment]
Can you share your code so that I can reproduce the error and debug it?

**Reply by JohannesMaierhofer** [reply]
While crafting the minimal example, I believe I've identified a solution. By implementing a `return True` in cases where no data needs to be transmitted back to the client, and using `return "{}", 200`, I successfully resolved the error warning. Although I am uncertain if this approach adheres to the official guidelines, it effectively addresses the issue in my particular context.

**Reply by miguelgrinberg** [reply]
What did you return that caused the error?

**Reply by JohannesMaierhofer** [reply]
indeed just `True`

**Reply by miguelgrinberg** [reply, marked as answer]
You returned true from the view function? That is certainly a problem. Valid returns are string, bytes or an iterable that returns strings or bytes.
